src/parser/Expression.ts

```diff
@@ -939,6 +939,56 @@ export class NewExpression extends Expression {
     }
 }
 
+/**
+ * This expression changes a call expression into a createObject call,
+ * eg `new roSGNode("Poster")` transpiles to `CreateObject("roSGNode", "Poster")`
+ */
+export class NewCreateObjectExpression extends Expression {
```

Comment:
Let's not create a separate statement for this. At the end of the day, it's still just a `NewExpression` semantically.

Reply:
Hmm... What about using NewExpression, but having a public flag on that class that says if this `isCreateObjectCall` or something?
We still want to be able to tell external to the new expression if it's for `createObject` for validation, etc.

Reply:
In the future, the entire list of those `ro*` components will be supplied from type definitions, some of which might be user-defined (to support if Roku has super-secret components we're not allowed to know about), so we won't know at parse time what the item should resolve to. Plus, if we eventually support something like `new Poster()`, you definitely won't know at parse time because it could be a user-defined component. 

With that in mind, how would you be able to create the flag? What if the component we're calling `new` on hasn't been parsed yet?